Activities/Cryptography/UiPath.Cryptography.Activities/KeyedHashFile.cs

```diff
@@ -92,34 +99,44 @@ protected override string Execute(CodeActivityContext context)
 
             try
             {
-                string filePath = FilePath.Get(context);
-                string key = Key.Get(context);
-                SecureString keySecureString = KeySecureString.Get(context);
-                Encoding keyEncoding = Encoding.Get(context);
+                var filePath = FilePath.Get(context);
+                var key = Key.Get(context);
+                var keySecureString = KeySecureString.Get(context);
+                var keyEncoding = Encoding.Get(context);
+                var inputFile = InputFile.Get(context);
 
-                if (string.IsNullOrWhiteSpace(filePath))
-                {
+                if (string.IsNullOrWhiteSpace(filePath) && inputFile == null)
                     throw new ArgumentNullException(Resources.FilePathDisplayName);
-                }
+
                 if (string.IsNullOrWhiteSpace(key) && keySecureString == null)
-                {
                     throw new ArgumentNullException(Resources.KeyAndSecureStringNull);
-                }
+
                 if (key != null && keySecureString != null)
-                {
                     throw new ArgumentNullException(Resources.KeyAndSecureStringNotNull);
-                }
+
                 if (keyEncoding == null)
-                {
                     throw new ArgumentNullException(Resources.Encoding);
-                }
-                if (!File.Exists(filePath))
-                {
+
+                if (!File.Exists(filePath) && inputFile == null)
                     throw new ArgumentException(Resources.FileDoesNotExistsException, Resources.FilePathDisplayName);
-                }
 
+                byte[] hashed;
+
+                if (inputFile != null && !inputFile.IsFolder)
+                {
+                    // Get local file
+                    var localFile = inputFile.ToLocalResource();
+                    //Run Sync
+                    Task.Run(async () => await localFile.ResolveAsync()).GetAwaiter().GetResult();
```

Comment:
These ifs still need consolidating. No need to call CryptographyHelper.HashDataWithKey both in the if and in the else branch, refactor so that it's called only once after the if..else block. See how it's already done in this PR for other activities.

Reply:
fixed